# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424201846 18000
#      Tue Feb 17 14:37:26 2015 -0500
# Node ID 9f6ee9cb749c48346e18140a78d31c517e29045c
# Parent  08033d58343501770f7cf188daed52530547d726
put OSMESA compiler flag variables in config_options struct

* common.mk, toplev.cc, oct-conf.in.h: Put OSMESA compiler flag
variables in config_options struct.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -259,16 +259,20 @@ LLVM_LDFLAGS = @LLVM_LDFLAGS@
 LLVM_LIBS = @LLVM_LIBS@
 
 MAGICK_CPPFLAGS = @MAGICK_CPPFLAGS@
 MAGICK_LDFLAGS = @MAGICK_LDFLAGS@
 MAGICK_LIBS = @MAGICK_LIBS@
 
 OPENGL_LIBS = @OPENGL_LIBS@
 
+OSMESA_CPPFLAGS = @OSMESA_CPPFLAGS@
+OSMESA_LDFLAGS = @OSMESA_LDFLAGS@
+OSMESA_LIBS = @OSMESA_LIBS@
+
 PCRE_CPPFLAGS = @PCRE_CPPFLAGS@
 PCRE_LIBS = @PCRE_LIBS@
 
 PTHREAD_CFLAGS = @PTHREAD_CFLAGS@
 PTHREAD_LIBS = @PTHREAD_LIBS@
 
 QHULL_CPPFLAGS = @QHULL_CPPFLAGS@
 QHULL_LDFLAGS = @QHULL_LDFLAGS@
@@ -603,16 +607,19 @@ echo "making $@ from $<"
   -e 's|%OCTAVE_CONF_MKOCTFILE_RANLIB%|\"${MKOCTFILE_RANLIB}\"|' \
   -e "s|%OCTAVE_CONF_OCTAVE_LINK_DEPS%|\"${OCTAVE_LINK_DEPS}\"|" \
   -e "s|%OCTAVE_CONF_OCTAVE_LINK_OPTS%|\"${OCTAVE_LINK_OPTS}\"|" \
   -e "s|%OCTAVE_CONF_OCTINCLUDEDIR%|\"${octincludedir}\"|" \
   -e "s|%OCTAVE_CONF_OCTLIBDIR%|\"${octlibdir}\"|" \
   -e "s|%OCTAVE_CONF_OCT_LINK_DEPS%|\"${OCT_LINK_DEPS}\"|" \
   -e "s|%OCTAVE_CONF_OCT_LINK_OPTS%|\"${OCT_LINK_OPTS}\"|" \
   -e "s|%OCTAVE_CONF_OPENGL_LIBS%|\"${OPENGL_LIBS}\"|" \
+  -e "s|%OCTAVE_CONF_OSMESA_CPPFLAGS%|\"${OSMESA_CPPFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_OSMESA_LDFLAGS%|\"${OSMESA_LDFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_OSMESA_LIBS%|\"${OSMESA_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_PCRE_CPPFLAGS%|\"${PCRE_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_PCRE_LIBS%|\"${PCRE_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_PREFIX%|\"${prefix}\"|" \
   -e "s|%OCTAVE_CONF_PTHREAD_CFLAGS%|\"${PTHREAD_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_PTHREAD_LIBS%|\"${PTHREAD_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_CPPFLAGS%|\"${QHULL_CPPFLAGSS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_LDFLAGS%|\"${QHULL_LDFLAGSS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_LIBS%|\"${QHULL_LIBS}\"|" \
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -1379,16 +1379,19 @@ specified option.\n\
     { false, "MAGICK_LDFLAGS", OCTAVE_CONF_MAGICK_LDFLAGS },
     { false, "MAGICK_LIBS", OCTAVE_CONF_MAGICK_LIBS },
     { false, "MKOCTFILE_DL_LDFLAGS", OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS },
     { false, "OCTAVE_LINK_DEPS", OCTAVE_CONF_OCTAVE_LINK_DEPS },
     { false, "OCTAVE_LINK_OPTS", OCTAVE_CONF_OCTAVE_LINK_OPTS },
     { false, "OCT_LINK_DEPS", OCTAVE_CONF_OCT_LINK_DEPS },
     { false, "OCT_LINK_OPTS", OCTAVE_CONF_OCT_LINK_OPTS },
     { false, "OPENGL_LIBS", OCTAVE_CONF_OPENGL_LIBS },
+    { false, "OSMESA_CPPFLAGS", OCTAVE_CONF_OSMESA_CPPFLAGS },
+    { false, "OSMESA_LDFLAGS", OCTAVE_CONF_OSMESA_LDFLAGS },
+    { false, "OSMESA_LIBS", OCTAVE_CONF_OSMESA_LIBS },
     { false, "PCRE_CPPFLAGS", OCTAVE_CONF_PCRE_CPPFLAGS },
     { false, "PCRE_LIBS", OCTAVE_CONF_PCRE_LIBS },
     { false, "PTHREAD_CFLAGS", OCTAVE_CONF_PTHREAD_CFLAGS },
     { false, "PTHREAD_LIBS", OCTAVE_CONF_PTHREAD_LIBS },
     { false, "QHULL_CPPFLAGS", OCTAVE_CONF_QHULL_CPPFLAGS },
     { false, "QHULL_LDFLAGS", OCTAVE_CONF_QHULL_LDFLAGS },
     { false, "QHULL_LIBS", OCTAVE_CONF_QHULL_LIBS },
     { false, "QRUPDATE_CPPFLAGS", OCTAVE_CONF_QRUPDATE_CPPFLAGS },
diff --git a/libinterp/oct-conf.in.h b/libinterp/oct-conf.in.h
--- a/libinterp/oct-conf.in.h
+++ b/libinterp/oct-conf.in.h
@@ -453,16 +453,28 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_OCT_LINK_OPTS
 #define OCTAVE_CONF_OCT_LINK_OPTS %OCTAVE_CONF_OCT_LINK_OPTS%
 #endif
 
 #ifndef OCTAVE_CONF_OPENGL_LIBS
 #define OCTAVE_CONF_OPENGL_LIBS %OCTAVE_CONF_OPENGL_LIBS%
 #endif
 
+#ifndef OCTAVE_CONF_OSMESA_CPPFLAGS
+#define OCTAVE_CONF_OSMESA_CPPFLAGS %OCTAVE_CONF_OSMESA_CPPFLAGS%
+#endif
+
+#ifndef OCTAVE_CONF_OSMESA_LDFLAGS
+#define OCTAVE_CONF_OSMESA_LDFLAGS %OCTAVE_CONF_OSMESA_LDFLAGS%
+#endif
+
+#ifndef OCTAVE_CONF_OSMESA_LIBS
+#define OCTAVE_CONF_OSMESA_LIBS %OCTAVE_CONF_OSMESA_LIBS%
+#endif
+
 #ifndef OCTAVE_CONF_PCRE_CPPFLAGS
 #define OCTAVE_CONF_PCRE_CPPFLAGS %OCTAVE_CONF_PCRE_CPPFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_PCRE_LIBS
 #define OCTAVE_CONF_PCRE_LIBS %OCTAVE_CONF_PCRE_LIBS%
 #endif
 
